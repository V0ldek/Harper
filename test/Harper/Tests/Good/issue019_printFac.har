facPrintSteps :: Integer -> sideeffect -> Integer;
facPrintSteps n = {
  var (result :: Integer) = 1;
  var (i :: Integer) = 0;
  while i < n {
    i += 1;
    result *= i;
    eval printLn result ();
  }
  return result;
};

main :: sideeffect -> ();
main = {
  (f :: sideeffect -> Integer) = facPrintSteps 3;
  eval printLn "After f." ();
  (n :: Integer) = f ();
  eval print "n = " ();
  eval printLn n ();
  eval f ();
};