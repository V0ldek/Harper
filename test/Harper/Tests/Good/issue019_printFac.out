After f.
1
2
6
n = 6
1
2
6

Execution ended with value:
()